ddtrace/contrib/grpc/client_interceptor.py

```diff
@@ -1,61 +1,82 @@
+import collections
 import grpc
 
-from ddtrace import Pin
-from .propagation import inject_span
+from ddtrace import config
+from ...propagation.http import HTTPPropagator
 from ...constants import ANALYTICS_SAMPLE_RATE_KEY
-from ...settings import config
 
+# DEV: Follows Python interceptors RFC laid out in
+# https://github.com/grpc/proposal/blob/master/L13-python-interceptors.md
 
-class GrpcClientInterceptor(
-        grpc.UnaryUnaryClientInterceptor, grpc.UnaryStreamClientInterceptor,
-        grpc.StreamUnaryClientInterceptor, grpc.StreamStreamClientInterceptor):
-    """Intercept calls on a channel. It creates span as well as doing the propagation
-    Derived from https://github.com/grpc/grpc/blob/d0cb61eada9d270b9043ec866b55c88617d362be/examples/python/interceptors/headers/generic_client_interceptor.py#L19
-    """  # noqa
-
-    def __init__(self, host, port):
-        self._pin = Pin.get_from(grpc)
-        self._host = host
-        self._port = port
-
-    def _start_span(self, method):
-        span = self._pin.tracer.trace('grpc.client', span_type='grpc', service=self._pin.service, resource=method)
-        span.set_tag('grpc.host', self._host)
-        if (self._port is not None):
-            span.set_tag('grpc.port', self._port)
-        if self._pin.tags:
-            span.set_tags(self._pin.tags)
-        # set analytics sample rate
-        span.set_tag(
-            ANALYTICS_SAMPLE_RATE_KEY,
-            config.grpc.get_analytics_sample_rate()
-        )
-        return span
 
-    def intercept_unary_unary(self, continuation, client_call_details, request):
-        return self.intercept_unary_stream(continuation, client_call_details, request)
+def create_client_interceptor(pin):
+    def interceptor_function(continuation, client_call_details,
+                             request_or_iterator):
+        if not pin.enabled:
+            response = continuation(client_call_details, request_or_iterator)
+            return response
+
+        with pin.tracer.trace(
+                '{}.client'.format(pin.app),
```

Comment:
why rely on `pin.app` here? Seems like this should always be `grpc.client`

(also, I think we want to remove `pin.app`)